File: backend_dart/bin/query.dart
import 'dart:io';
import 'package:dart_firebase_admin/firestore.dart';
import 'package:backend_dart/config/env.dart';
import 'package:backend_dart/config/firebase.dart';

Future<void> main() async {
  env.load();
  await FirebaseConfig.initialize();
  
  print('Connected to Firestore');
  final snapshot = await FirebaseConfig.db!.collection('users').get();
  final out = StringBuffer();
  out.writeln('Found ${snapshot.docs.length} users');
  
  for (final doc in snapshot.docs) {
    final data = doc.data();
    out.writeln('${doc.id} | email=${data['email']} | name=${data['name']}');
  }

  File('users_out.txt').writeAsStringSync(out.toString());
  print('Done typing to users_out.txt');

  exit(0);
}

File: backend_dart/query.js
const admin = require('firebase-admin');
const fs = require('fs');

const env = fs.readFileSync('.env', 'utf8');
const match = env.match(/FIREBASE_SERVICE_ACCOUNT_BASE64=(.+)/);
const serviceAccountString = Buffer.from(match[1], 'base64').toString('utf8');
const serviceAccount = JSON.parse(serviceAccountString);

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();

async function run() {
  const snapshot = await db.collection('vaultItems').get();
  console.log(`Found ${snapshot.size} items in vaultItems`);
  snapshot.forEach(doc => {
    const data = doc.data();
    console.log(`${data.name} | user=${data.user} | type=${data.type}`);
  });
  
  const snapshot2 = await db.collection('vault_items').get();
  console.log(`Found ${snapshot2.size} items in vault_items`);
  snapshot2.forEach(doc => {
    const data = doc.data();
    console.log(`${data.name} | user=${data.user} | type=${data.type}`);
  });
}

run().catch(console.error);
